chain/store/store.go

```diff
@@ -860,7 +860,13 @@ func NewChainRand(cs *ChainStore, blks []cid.Cid, bheight uint64, tickets []*typ
 	}
 }
 
+// GetRandomness is to get ticket from the epoch at height h
+// Note that h might be a negative number, e.g. when get a randomness for sector precommit
 func (cr *chainRand) GetRandomness(ctx context.Context, h int64) ([]byte, error) {
 	lb := (int64(cr.bh) + int64(len(cr.tickets))) - h
-	return cr.cs.GetRandomness(ctx, cr.blks, cr.tickets, lb)
+	if lb < 0 {
```

Comment:
this would imply that `h` is in the future, aka, `h > cr.bh + len(cr.tickets)`. Maybe the error message should say something like "cannot return randomness from the future"?

Reply:
Solved with the new commit: ea6e0fd